### PR TITLE
[SPARK-49313][BUILD][TESTS] Upgrade `DB2` & `MySQL` & `Postgres` & `Mariadb` docker image version

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to upgrade some db docker image version, include:
- `DB2` from `11.5.8.0` to `11.5.9.0` (NOTE: Its namespace (from `ibmcom/db2` to `icr.io/db2_community/db2`) has been changed, see: https://hub.docker.com/r/ibmcom/db2)
  image
- `MySQL` from `9.0.0` to `9.0.1`
- `Mariadb` from `10.5.25` to `10.5.26`
- `Postgres` from `16.3-alpine` to `16.4-alpine`

   
### Why are the changes needed?
Tests dependencies upgrading.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @yaooqinn @dongjoon-hyun @HyukjinKwon

File: connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/DB2DatabaseOnDocker.scala [hunk 1]
```diff
@@ -18,7 +18,8 @@
 package org.apache.spark.sql.jdbc
 
 class DB2DatabaseOnDocker extends DatabaseOnDocker {
-  override val imageName = sys.env.getOrElse("DB2_DOCKER_IMAGE_NAME", "ibmcom/db2:11.5.8.0")
+  override val imageName = sys.env.getOrElse("DB2_DOCKER_IMAGE_NAME",
+    "icr.io/db2_community/db2:11.5.9.0")
```
Comment (dongjoon-hyun) [inline connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/DB2DatabaseOnDocker.scala:22]: Thank you for this update.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you, @panbingkun .

**Pull request closed without merging**